src/main/java/ch/njol/skript/conditions/CondIsOnline.java

```diff
@@ -29,36 +29,43 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Is Online")
-@Description("Checks whether a player is online.")
-@Examples({"player is online",
-		"player-argument is offline"})
+@Description("Checks whether a player is online. The 'connected' pattern will return false once this player leaves the server, even if they rejoin.")
+@Examples({
+	"player is online",
+	"player-argument is offline",
+	"while player is connected:",
+	"\twait 60 seconds",
+	"\tsend \"hello!\" to player"
+})
 @Since("1.4")
 public class CondIsOnline extends PropertyCondition<OfflinePlayer> {
 	
 	static {
-		register(CondIsOnline.class, "(online|1¦offline)", "offlineplayers");
+		register(CondIsOnline.class, "(online|:offline|:connected)", "offlineplayers");
 	}
 	
+	private boolean connected; // https://github.com/SkriptLang/Skript/issues/6100
+	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
```

Comment:
While we're at it \:)
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```